app/ts/library/ethereum.ts

```diff
@@ -0,0 +1,32 @@
+import { BigNumber, ethers } from 'ethers'
+import { Result } from 'ethers/lib/utils.js'
+import { TransactionReceipt, TransactionResponse, TransferTransactionResponse } from '../types.js'
+import { ERC20ABI } from './ERC20ABI.js'
+
+export function isTransferTransaction(txResponse: TransactionResponse): txResponse is TransferTransactionResponse {
+	return txResponse.data.toLowerCase().startsWith('0xa9059cbb')
+}
+
+export function getTransferTokenValue(transactionReceipt: TransactionReceipt) {
+	const erc20Interface = new ethers.utils.Interface(ERC20ABI)
+	const transferLog = transactionReceipt.logs.find(isTransferLog)
```

Comment:
I'm still a bit uncomfortable with this.  A single transaction can have multiple transfer events.  For example, some ERC20 tokens will transfer a fee to the token creator every time a token is transferred, as well as transferring to the designated recipient.  In this code we are blindly assuming that the first transfer event (log) is the one we care about, but there is no guarantee in the standard that that is the case.

Do we know who the expected sender/recipient is when we are calling this?  If so, we should filter on those.  If we don't, we may want to consider just presenting all of the events to the user rather than just the first one.

Note: If it is a lot of work to do these, we can file issues for them and address them in a separate PR.

Reply:
Yes I agree, should definitely find all matching transfer.